components/director/internal/systemfetcher/loader.go

```diff
@@ -0,0 +1,212 @@
+	files, err := readDir(integrationSystemsDirectoryPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while reading directory with integration systems files [%s]", integrationSystemsDirectoryPath)
+	}
+
+	var integrationSystems []model.IntegrationSystem
+	for _, f := range files {
+		log.C(ctx).Infof("Loading integration systems from file: %s", f.Name())
+
+		if filepath.Ext(f.Name()) != ".json" {
+			return nil, apperrors.NewInternalError(fmt.Sprintf("unsupported file format %q, supported format: json", filepath.Ext(f.Name())))
+		}
+
+		bytes, err := readFile(integrationSystemsDirectoryPath + f.Name())
+		if err != nil {
+			return nil, errors.Wrapf(err, "while reading integration systems file %q", integrationSystemsDirectoryPath+f.Name())
+		}
+
+		var integrationSystemsFromFile []model.IntegrationSystem
+		if err := json.Unmarshal(bytes, &integrationSystemsFromFile); err != nil {
+			return nil, errors.Wrapf(err, "while unmarshalling integration systems from file %s", integrationSystemsDirectoryPath+f.Name())
+		}
+		log.C(ctx).Infof("Successfully loaded integration systems from file: %s", f.Name())
+		integrationSystems = append(integrationSystems, integrationSystemsFromFile...)
+	}
+
+	return integrationSystems, nil
+}
+
+func (d *DataLoader) loadAppTemplates(ctx context.Context, readDir func(dirname string) ([]fs.FileInfo, error), readFile func(filename string) ([]byte, error)) ([]model.ApplicationTemplateInput, error) {
+	files, err := readDir(applicationTemplatesDirectoryPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while reading directory with application templates files [%s]", applicationTemplatesDirectoryPath)
+	}
+
+	var appTemplateInputs []model.ApplicationTemplateInput
+	for _, f := range files {
+		log.C(ctx).Infof("Loading application templates from file: %s", f.Name())
+
+		if filepath.Ext(f.Name()) != ".json" {
+			return nil, apperrors.NewInternalError(fmt.Sprintf("unsupported file format %q, supported format: json", filepath.Ext(f.Name())))
+		}
+
+		bytes, err := readFile(applicationTemplatesDirectoryPath + f.Name())
+		if err != nil {
+			return nil, errors.Wrapf(err, "while reading application templates file %q", applicationTemplatesDirectoryPath+f.Name())
+		}
+
+		var templatesFromFile []model.ApplicationTemplateInput
+		if err := json.Unmarshal(bytes, &templatesFromFile); err != nil {
+			return nil, errors.Wrapf(err, "while unmarshalling application templates from file %s", applicationTemplatesDirectoryPath+f.Name())
+		}
+		log.C(ctx).Infof("Successfully loaded application templates from file: %s", f.Name())
+		appTemplateInputs = append(appTemplateInputs, templatesFromFile...)
+	}
+
+	return appTemplateInputs, nil
+}
+
+func (d *DataLoader) upsertIntegrationSystems(ctx context.Context, intSystems []model.IntegrationSystem) error {
```

Comment:
This upsetting func and the one for app templates only create the respective entities if they are missing. From this I think it would be more accurate to name the function'insertIntegrationsSystemsIfMissing' as there won't be any updating of the entity if it already exists.

Reply:
Would it be much effort to also update the entity if it already exists with the provided configuration? In this way we would truly have one configuration which controls completely the state of predefined app templates in our db.

Reply:
On second thoughts:
- we don't have much use case of updating int systems as they have no meaningful data associated with them
- as far as app templates are concerned - if you look at my comment below about restructuring the templates.json array where the only two properties that would make sense are `name` and `applicationInputJSON` then updating and existing apptemplate here would mean only to update it's applicationInputJSON property as is provided in the configuration.